CMakeLists.txt

```diff
@@ -1,6 +1,10 @@
 cmake_minimum_required(VERSION 3.5)
 project (Calico)
 
+# set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_CURRENT_LIST_DIR}/build/lib)
+# set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_CURRENT_LIST_DIR}/build/lib)
+# set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_LIST_DIR}/build/bin)
```

Comment:
if you uncomment these lines, the core code will build in (root:/build), and the python code will leverage that code (so you could stack/cache core first, then build the python code without re-compiling).

The problem I was running into is these set() calls seem to be global scope, so the Python code then doesn't compile into the directories pybind11 is expecting, (it actually goes into root:/build/ too) and then the wheel is empty.

I think the solution is to scope these set()s so they only apply here, not to the calling CMake, but I'm not sure how to set that up.